src/helpers/figures/charts/chart_ui_common.ts

```diff
@@ -50,6 +50,33 @@ export function filterEmptyDataPoints(
   };
 }
 
+export function aggregateDataForLabels(
+  labels: string[],
+  datasets: DatasetValues[]
+): { labels: string[]; dataSetsValues: DatasetValues[] } {
+  const parseNumber = (value) => (typeof value === "number" ? value : 0);
+
+  const labelMap = {};
+  for (let indexOfLabel = 0; indexOfLabel < labels.length; ++indexOfLabel) {
+    const label = labels[indexOfLabel];
+    if (!labelMap[label]) {
+      labelMap[label] = datasets.map((dataset) => parseNumber(dataset.data[indexOfLabel]));
+      continue;
```

Comment:
I would initialize `labelMap` for all labels first, with zeros.
And only then fill it with the values.
That would avoid this nested `if` and the `continue`